google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -43,35 +43,93 @@ void TracingMessageBatch::SaveMessage(pubsub::Message m) {
 
 namespace {
 
-opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeParentSpan(
+/// Inserts a link for each span in @p destination between the @p begin and @p
+/// end iterators.
```

Comment:
Perhaps `Inserts a link for each span between @p begin and @p end into @p destination.`

But that said, "between" is always problematic.  It might be best to use terminology like `in the range @p begin to @end` to piggy-back off the common knowledge of "range" meaning an open-ended interval.